### PR TITLE
Address s3-test functional test suite failures

### PR DESCRIPTION
**Pull request by kellymclaughlin**
This pull request addresses the failures from #577. It requires related changes to `webmachine`, `mochiweb`, and the Basho fork of `s3-tests`.

Related PRs:

https://github.com/basho/mochiweb/pull/10
https://github.com/basho/webmachine/pull/150
https://github.com/basho/webmachine/pull/144
https://github.com/basho/s3-tests/pull/1

I have created a `rebar.config` that could be used to test these in concert if using a `rebar` built with the changes outlined in [this](https://github.com/rebar/rebar/pull/92) PR. 

```
{sub_dirs, ["rel"]}.

{require_otp_vsn, "R14B0[34]|R15"}.

{cover_enabled, true}.

{lib_dirs, ["deps", "apps"]}.

{erl_opts, [debug_info, warnings_as_errors, {parse_transform, lager_transform}]}.

{reset_after_eunit, true}.

{plugin_dir, ".plugins"}.
{plugins, [rebar_test_plugin]}.
{client_test, [
    {test_paths, ["client_tests/erlang"]},
    {test_output, ".client_test"}
]}.
{int_test, [
    {test_paths, ["int_test"]},
    {test_output, ".int_test"}
]}.
{riak_test, [
    {test_paths, ["riak_test/tests", "riak_test/src"]},
    {test_output, "riak_test/ebin"}
]}.

{deps, [
        {node_package, "1.2.2", {git, "git://github.com/basho/node_package", {tag, "1.2.2"}}},
    {mochiweb, ".*", {git, "git://github.com/basho/mochiweb", {branch, "klm-strip-header-values-take2"}}, [force_version]},
    {webmachine, ".*", {git, "git://github.com/basho/webmachine", {branch, "klm-handle-empty-content-type-rebase-testing-only"}}, [force_version]},
        {riakc, ".*", {git, "git://github.com/basho/riak-erlang-client", {tag, "1.3.1.1"}}},
        {lager, ".*", {git, "git://github.com/basho/lager", {tag, "1.2.2"}}},
        {eper, ".*", {git, "git://github.com/basho/eper.git", "3280b736057a6cf5c01590c79fd192f7e8645270"}},
        {sext, ".*", {git, "git://github.com/esl/sext", {tag, "0.4.1"}}},
        {druuid, ".*", {git, "git://github.com/kellymclaughlin/druuid.git", {tag, "0.2"}}},
        {velvet, "1.3.*", {git, "git://github.com/basho/velvet", {tag, "1.3.0"}}},
        {poolboy, "0.8.1", {git, "git://github.com/basho/poolboy", "0e15b5dcbff89b3d1d4021591f90375d4d2ee9a1"}},
        {folsom, ".*", {git, "git://github.com/boundary/folsom", {tag, "0.7.4"}}},
        {erlcloud, ".*", {git, "git://github.com/basho/erlcloud.git", {tag, "0.4.1p1"}}}
       ]}.
```

The `klm-handle-empty-content-type-rebase-testing-only` branch of webmachine is a branch I created just for this testing that combines the change sets from the two webmachine PRs mentioned above.

### CODE REVIEW

**Comment by reiddraper** [comment]
Missing gitignore for `client_tests/python/ceph_tests/s3-tests/`

**Comment by reiddraper** [comment]
Getting this error trying to get the deps, maybe not pushed to github? Using your build of rebar.

```
➜  riak_cs git:(bugfix/s3-test-function-suite-failures) ✗ $basho/rebar/rebar get-deps
==> rel (get-deps)
==> riak_cs (get-deps)
Pulling node_package from {git,"git://github.com/basho/node_package",
                               {tag,"1.2.2"}}
Cloning into 'node_package'...
Pulling mochiweb from {git,"git://github.com/basho/webmachine",
                           {branch,"klm-strip-header-values-take2"}}
Cloning into 'mochiweb'...
error: pathspec 'origin/klm-strip-header-values-take2' did not match any file(s) known to git.
ERROR: git checkout -q origin/klm-strip-header-values-take2 failed with error: 1 and output:
error: pathspec 'origin/klm-strip-header-values-take2' did not match any file(s) known to git.

ERROR: 'get-deps' failed while processing /Users/reid/Documents/repos/basho/riak_cs: rebar_abort
```

**Comment by kellymclaughlin** [comment]
Looks like I flubbed the repo spec and have it pointing to webmachine
instead of mochiweb.

**Comment by reiddraper** [comment]
> Looks like I flubbed the repo spec and have it pointing to webmachine instead of mochiweb.

Perfect, that got me running

**Comment by reiddraper** [comment]
Hmm, getting some failures running the test now. Will give it a shot again on my flight tomorrow AM.

**Comment by reiddraper** [comment]
Riak CS commit: 6069910ceb07819f38f5bfa22ece10066eb28248

Here are my deps:

``` bash
for D in `ls deps`; do cd deps/$D; echo $D `git rev-parse HEAD`; cd ../..; done
bear 3fd09d1b7bbd9de5b2d29f46df04a93fca9ce85e
druuid b3c5c2a52bb3f510d168b32e64f6fbc6a3c6a0e6
edown 1f6954ce8f313139385110f68c18c17adb2723f8
eper 3280b736057a6cf5c01590c79fd192f7e8645270
erlcloud b83090b5219c628369ece0df96ede5d02d44b17d
folsom 154024e3c4f52b06a33bcb6b901bcf0917e80fc4
getopt 659a28f4145bc9843598972854299dc4ea77e4cb
lager 4fcb8fd40f6ab7a3fa30f4d5b559a10788ac7886
meck 161e240166a0d20e9ffa2be5d0bfa6a981f12aef
mochiweb d19ec05f4bffdfabe856bc9b8a6bbdc32d484f5c
node_package 6323ccbab12ab2b81ad95f5e285fafbbe396da3b
poolboy 0e15b5dcbff89b3d1d4021591f90375d4d2ee9a1
protobuffs 839e441c194fb946b2a72df24252b377269169cb
riak_pb 033c7843edfa0960ee3300a5387809ff8393dd9d
riak_repl_pb_api ad6ca80ff1fcfe62a12fa4485ceed64e86603cfe
riakc 444feec6d81a8a382b452d0f127cdcb3eac1671c
sext 362bdd178307ab531a7b78dd39de37f9e91108ef
velvet 72e83d0ec40a93b78c6cc764dbc017023b6e31f5
webmachine 17e30e1334e68fb09fde2d90158faee79fbb565b
```

and the errors I see:

Riak CS console:

```
00:13:52.584 [error] webmachine error: path="/buckets"
"Method Not Allowed"
[{xmerl_xml,'Error',[[[["<","Code",">"],["InvalidBucketName"],["</","Code",">"]],[["<","Message",">"],["The specified bucket is not valid."],["</","Message",">"]],[["<","Resource",">"],["/_alphasoup"],["</","Resource",">"]],[["<","RequestId",">"],[[]],["</","RequestId",">"]]],[],[],{xmlElement,'Error',[],[],{xmlNamespace,[],[]},[],1,[],[{xmlElement,'Code',[],[],{xmlNamespace,[],[]},[{'Error',1}],1,[],[{xmlText,[{'Code',1},{'Error',1}],1,[],"InvalidBucketName",text}],[],[],undeclared},{xmlElement,'Message',[],[],{xmlNamespace,[],[]},[{'Error',1}],2,[],[{xmlText,[{'Message',2},{'Error',1}],1,[],"The specified bucket is not valid.",text}],[],[],undeclared},{xmlElement,'Resource',[],[],{xmlNamespace,[],[]},[{'Error',1}],3,[],[{xmlText,[{'Resource',3},{'Error',1}],1,[],"/_alphasoup",text}],[],[],undeclared},{xmlElement,'RequestId',[],[],{xmlNamespace,[],[]},[{'Error',1}],4,[],[{xmlText,[{'RequestId',4},{'Error',1}],1,[],[],text}],[],[],undeclared}],[],[],undeclared}],[]},{xmerl,apply_cb,5,[{file,"xmerl.erl"},{line,309}]},{xmerl,tagdef,5,[{file,"xmerl.erl"},{line,259}]},{xmerl,export_content,2,[{file,"xmerl.erl"},{line,197}]},{xmerl,export1,3,[{file,"xmerl.erl"},{line,166}]},{riak_cs_xml,export_xml,1,[{file,"src/riak_cs_xml.erl"},{line,54}]},{riak_cs_s3_response,error_response,5,[{file,"src/riak_cs_s3_response.erl"},{line,202}]},{riak_cs_wm_common,accept_body,2,[{file,"src/riak_cs_wm_common.erl"},{line,337}]}]
```

[Tests output](https://gist.github.com/reiddraper/fe95b07b087e9d27d178)

**Comment by kellymclaughlin** [comment]
I see the same output on the console, but i get all tests passing using a fresh clone and build of riak_cs. Did you remember to check out the `klm-proxy-support` branch of the `s3-tests` repo? I will take a look at those console messages, particularly the XML error.

**Comment by reiddraper** [comment]
> Did you remember to check out the klm-proxy-support branch of the s3-tests repo? 

I thought I had, but lemme double check

**Comment by reiddraper** [comment]
Welp, yay state. Dunno why this fixed it, but some combo of `make clean compile stage`, `make clean` in the python dir and then re-checking out the s3-tests branch fixed it. Maybe I hadn't successfully checked out the branch the first time? Who knows, but yay, it passes now!!

**Comment by kellymclaughlin** [comment]
I did push some more fixes to `s3-tests` late this afternoon so that could have been the difference.

**Comment by reiddraper** [comment]
Everything working and passing for me. Going to take a look at the actual changes and related PRs now.

File: src/riak_cs_wm_bucket.erl [hunk 1]
```diff
@@ -43,13 +43,16 @@ content_types_provided(RD, Ctx) ->
...
-    case wrq:get_req_header("content-type", RD) of
-        undefined ->
-            {[{"application/octet-stream", accept_body}], RD, Ctx};
-        CType ->
-            {Media, _Params} = mochiweb_util:parse_header(CType),
-            {[{Media, accept_body}], RD, Ctx}
-    end.
+    content_types_accepted(wrq:get_req_header("content-type", RD), RD, Ctx).
+
+-spec content_types_accepted(unndefined | string(), #wm_reqdata{}, #context{}) ->
```
Comment (reiddraper) [inline src/riak_cs_wm_bucket.erl]: s/unndefined/undefined

Comment (kellymclaughlin) [inline src/riak_cs_wm_bucket.erl]: Addressed by [9947798](https://github.com/basho/riak_cs/commit/99477985dd0b8f9bf40837e7a64bc9b2c53aa75f)

**Pull request merged**